src/asm.rs

```diff
@@ -26,13 +26,16 @@ pub fn bkpt() {
 
 /// Blocks the program for *at least* `n` instruction cycles
 ///
-/// This is implemented in assembly so its execution time is the same regardless of the optimization
-/// level.
+/// This is implemented in assembly so its execution time is independent of the optimization
+/// level, however it is dependent on the specific architecture and core configuration.
 ///
-/// NOTE that the delay can take much longer if interrupts are serviced during its execution.
+/// NOTE that the delay can take much longer if interrupts are serviced during its execution 
+/// and the execution time may wary with other factors. This delay is mainly useful for simple
+/// timer-less initialisation of peripherals iff accurate timing is not essential. In any other
```

Comment:
```suggestion
/// timer-less initialization of peripherals if accurate timing is not essential. In any other
```

Reply:
Hm, the **iff** is intended. I'm fine with changing the spelling to AE.

Reply:
Maybe expand it to "if and only if" then? *iff* tends to confuse people.

Reply:
Hm, I've never met anyone who was confused about that but sure, why not.